src/OpenFeature.Contrib.Providers.Flagsmith/FlagsmithProvider.cs

```diff
@@ -0,0 +1,207 @@
+﻿using Flagsmith;
+using OpenFeature.Constant;
+using OpenFeature.Model;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Threading.Tasks;
+using Trait = Flagsmith.Trait;
+
+namespace OpenFeature.Contrib.Providers.Flagsmith
+{
+    /// <summary>
+    /// FlagsmithProvider is the .NET provider implementation for the feature flag solution Flagsmith.
+    /// </summary>
+    public class FlagsmithProvider : FeatureProvider
+    {
+        private readonly static Metadata Metadata = new("Flagsmith Provider");
+        internal readonly IFlagsmithClient _flagsmithClient;
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="options">Flagsmith client options. You can just use <see cref="FlagsmithConfiguration"/> class</param>
+        public FlagsmithProvider(IFlagsmithConfiguration options)
+        {
+            _flagsmithClient = new FlagsmithClient(options);
+        }
+
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="options">Flagsmith client options. You can just use <see cref="FlagsmithConfiguration"/> class</param>
+        /// <param name="httpClient">Http client that will be used for flagsmith requests. You also can use it to register <see cref="FeatureProvider"/> as Typed HttpClient with <see cref="FeatureProvider"> as abstraction</see></param>
+        public FlagsmithProvider(IFlagsmithConfiguration options, HttpClient httpClient)
+        {
+            _flagsmithClient = new FlagsmithClient(options, httpClient);
+        }
+
+
+        /// <summary>
+        /// Creates new instance of <see cref="FlagsmithProvider"/>
+        /// </summary>
+        /// <param name="flagsmithClient">Precreated Flagsmith client. You can just use <see cref="FlagsmithClient"/> class.</param>
+        public FlagsmithProvider(IFlagsmithClient flagsmithClient)
+        {
+            _flagsmithClient = flagsmithClient;
+        }
+
+        private Task<IFlags> GetFlags(EvaluationContext ctx = null)
+        {
+            var key = ctx?.GetValue("targetingKey")?.AsString;
+            return string.IsNullOrEmpty(key)
+                ? _flagsmithClient.GetEnvironmentFlags()
+                : _flagsmithClient.GetIdentityFlags(key, ctx.AsDictionary().Select(x => new Trait(x.Key, x.Value.AsObject) as ITrait).ToList());
```

Comment:
I could have missed something, but I don't see tests for this logic.

I would expect to see at least one test that passes in an OpenFeature evaluation context, and asserts it's correctly converted to traits and `GetIdentityFlags` is called.

Reply:
You are right, i didn't add tests for this. I tried to replicate logic from java sdk, but have no idea how to create context for tests :(

Reply:
If you mean building an OpenFeature evaluation context, it's just a simple "bag" for keys/values. You could just add a few properties to it, and see if they are mapped correctly. [Here's an example test](https://github.com/open-feature/dotnet-sdk/blob/2cbdba80d836f8b7850e8dc5f1f1790ef2ed1aca/test/OpenFeature.Tests/OpenFeatureEvaluationContextTests.cs#L53-L59) which creates a context in the dotnet SDK.

I hope that helps. Let me know if it doesn;'t.

Reply:
I can try reflection

Reply:
added!